components/ButtonBar/examples/ButtonBarCustomizedFontExample.m

```diff
@@ -0,0 +1,117 @@
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "MaterialButtonBar.h"
+
+@interface ButtonBarCustomizedFontExample : UIViewController
+@end
+
+@implementation ButtonBarCustomizedFontExample
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  MDCButtonBar *buttonBar = [[MDCButtonBar alloc] init];
+
+  // MDCButtonBar ignores the style of UIBarButtonItem.
+  UIBarButtonItemStyle ignored = UIBarButtonItemStyleDone;
+
+  UIBarButtonItem *actionItem =
+      [[UIBarButtonItem alloc] initWithTitle:@"Disable"
+                                       style:ignored
+                                      target:self
+                                      action:@selector(didTapActionButton:)];
+  UIBarButtonItem *secondActionItem =
+      [[UIBarButtonItem alloc] initWithTitle:@"Customization"
+                                       style:ignored
+                                      target:self
+                                      action:@selector(didTapActionButton:)];
+
+  NSArray *items = @[ actionItem, secondActionItem ];
+
+  // Set the title text attributes before assigning to buttonBar.items
+  // because of https://github.com/material-components/material-components-ios/issues/277
+  for (UIBarButtonItem *item in items) {
+    [item setTitleTextAttributes:@{ NSForegroundColorAttributeName : [UIColor greenColor] }
+                        forState:UIControlStateNormal];
+  }
+
+  buttonBar.items = items;
+
+  // MDCButtonBar's sizeThatFits gives a "best-fit" size of the provided items.
+  CGSize size = [buttonBar sizeThatFits:self.view.bounds.size];
+  CGFloat x = (self.view.bounds.size.width - size.width) / 2;
+  CGFloat y = self.view.bounds.size.height / 2 - size.height;
+  buttonBar.frame = CGRectMake(x, y, size.width, size.height);
+  buttonBar.autoresizingMask =
+      (UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin |
+       UIViewAutoresizingFlexibleLeftMargin | UIViewAutoresizingFlexibleRightMargin);
+  [self.view addSubview:buttonBar];
+
+  // Ensure that the controller's view isn't transparent.
+  self.view.backgroundColor = [UIColor colorWithWhite:0.9f alpha:1.0f];
+}
+
+#pragma mark - User actions
+
+- (void)didTapActionButton:(id)sender {
+  NSLog(@"Did tap action item: %@", sender);
+
+  // Disable CUstomization
+  [[MDCButtonBarButton appearance] setTitleFont:nil forState:UIControlStateNormal];
```

Comment:
This won't produce any on-screen change. Can we set the sender's title back to `nil` to match the UIAppearance version? That way the UIAppearance value and on-screen value will be the same.

Reply:
I definitely appreciate the value of this, but the sender is actually a UIBarButtonItem.  The complexity of reseting the fonts currently used would be high.  UIAppearance properties seems like they should really should only be called once, ideally on App startup, so our samples are going to be non-standard.
I've added additional documentation to the sample detailing this.